src/apps/backend/modules/otp/internal/otp-writer.ts

```diff
@@ -0,0 +1,50 @@
+import { PhoneNumber } from '../../account';
+import {
+  Otp,
+  OtpExpiredError,
+  OtpIncorrectError,
+  OtpStatus,
+} from '../types';
+
+import OtpUtil from './otp-util';
+import OtpRepository from './store/otp-repository';
+
+export default class OtpWriter {
+  public static async createOtp(
+    phoneNumber: PhoneNumber,
+  ): Promise<Otp> {
+    const otpDb = await OtpRepository.create({
+      active: true,
+      otpCode: OtpUtil.generateOtp(),
+      phoneNumber,
+      status: OtpStatus.PENDING,
+    });
+
+    return OtpUtil.convertOtpDBToOtp(otpDb);
+  }
+
+  public static async verifyOTP(
+    phoneNumber: PhoneNumber,
+    otpCode: string,
+  ): Promise<Otp> {
+    const otpDb = await OtpRepository.findOne({
```

Comment:
Which one?

We want to allow user to resend OTP, so for a phone number you may have multiple OTPs

Reply:
Should we implement something like
expireAt: {
  type: Date,
  default: Date.now,
  index: { expires: '5m' },
},
in the schema?